src/ImageSharp.Web/DependencyInjection/ImageSharpBuilderExtensions.cs

```diff
@@ -377,7 +371,28 @@ public static IImageSharpBuilder Configure<TOptions>(this IImageSharpBuilder bui
              where TOptions : class
         {
             builder.Services.Configure(configureOptions);
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Registers an action used to configure a particular type of options. Note: These are run after all <see cref="Configure{TOptions}(IImageSharpBuilder, Action{TOptions})"/>.
+        /// </summary>
+        /// <typeparam name="TOptions">The options type to be configured.</typeparam>
+        /// <param name="builder">The core builder.</param>
+        /// <param name="configureOptions">The action used to configure the options.</param>
+        /// <returns>The <see cref="IImageSharpBuilder"/>.</returns>
+        public static IImageSharpBuilder PostConfigure<TOptions>(this IImageSharpBuilder builder, Action<TOptions> configureOptions)
+             where TOptions : class
+        {
+            builder.Services.PostConfigure(configureOptions);
+
             return builder;
         }
+
+        private static Type GetImplementationType(this ServiceDescriptor descriptor)
+            => descriptor.ImplementationType
+            ?? descriptor.ImplementationInstance?.GetType()
+            ?? descriptor.ImplementationFactory?.GetType().GenericTypeArguments[1];
```

Comment:
This is inspired by the [internal `ServiceDescriptor.GetImplementationType()` method](https://github.com/dotnet/runtime/blob/34f2b587f3512d3974d73b316e565194df34eea3/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ServiceDescriptor.cs#L95-L116).

Reply:
I think at the time I first started writing all this stuff. .NET Core 1 was still a thing and you needed `GetMethodInfo()`. This is much better.